### ISSUE TITLE
Search box triggers while typing `-` in a text input

### ISSUE THREAD

**Issue by gburning** [storybookjs/storybook#14441]
**Description**
When pressing `-` (hyphen) in a text input knob the search box is triggered.

Using version 6.1.21 and @storybook/addon-knobs of the same version.

I'm using a Swedish keyboard so the key that triggers `/` (forward slash) on english QWERTY keyboards instead writes a `-` (hyphen).

This keyboard layout is similar to the one I use (excluding the numpad):
http://kbdlayout.info/KBDSW/

The problem here is two-fold:
1. The shortcut should not trigger when I'm inputting text in any way.
2. The listed keyboard shortcut should ideally match the key I have to press.

The second problem is less important to me (even though it's confusing).

I should also note that writing `/` to trigger a keyboard shortcut, while fairly common in different apps, usually never works for me since I have to press the SHIFT+7 keys to write a `/`. So if the UI could be updated to match the key needed that is preferable.

**System**
```
  System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz
  Binaries:
    Node: 12.20.1 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.14.10 - /usr/local/bin/npm
  Browsers:
    Chrome: 89.0.4389.114
    Firefox: 83.0
    Safari: 13.1.3
```

**Comment by ghengeveld** [comment]
First problem should be resolved in 6.2: https://github.com/storybookjs/storybook/pull/14062

The second I don't really know how to reliably fix for any keyboard layout. We could use `code` rather than `key` so we target the *location* of the key rather than the `/` character, but that would probably be even more confusing, unless we somehow manage to render the actual character in the search box hint.

Please verify whether 6.2 resolves the issue for you.

**Comment by gburning** [comment]
The first problem is is fixed (tested with 6.2.7). Thanks for the quick turnaround!

As for the second problem, I agree. Maybe a better idea is to change which keys are used? I noticed that Docsearch uses cmd/ctrl+K. That seems to work well.

Regardless, the main problem is solved so feel free to close this issue.
